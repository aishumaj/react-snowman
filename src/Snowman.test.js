import { render, fireEvent } from "@testing-library/react"
import Snowman from "./Snowman"

test("photo displays after maximum guesses", () => {
  const { container, debug } = render(<Snowman words={["react"]} maxWrong={1} />);
  
  fireEvent.click(container.querySelector("#w"));
  
  //expect container to have last image
  const img = container.querySelector("img");
  
  expect(img.getAttribute("src")).toEqual("1.png");
  expect(container.querySelector(".letterButtons")).not.toBeInTheDocument();
 
})

test("matches snapshot", function () {
  const { container } = render(<Snowman />);
  expect(container).toMatchSnapshot();
});