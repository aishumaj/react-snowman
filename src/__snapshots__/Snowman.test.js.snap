// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="0"
      src="0.png"
    />
    <p>
      Wrong Guesses: 
      0
       hello
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p>
      <button
        class="letterButton"
        id="a"
        value="a"
      >
        a
      </button>
      <button
        class="letterButton"
        id="b"
        value="b"
      >
        b
      </button>
      <button
        class="letterButton"
        id="c"
        value="c"
      >
        c
      </button>
      <button
        class="letterButton"
        id="d"
        value="d"
      >
        d
      </button>
      <button
        class="letterButton"
        id="e"
        value="e"
      >
        e
      </button>
      <button
        class="letterButton"
        id="f"
        value="f"
      >
        f
      </button>
      <button
        class="letterButton"
        id="g"
        value="g"
      >
        g
      </button>
      <button
        class="letterButton"
        id="h"
        value="h"
      >
        h
      </button>
      <button
        class="letterButton"
        id="i"
        value="i"
      >
        i
      </button>
      <button
        class="letterButton"
        id="j"
        value="j"
      >
        j
      </button>
      <button
        class="letterButton"
        id="k"
        value="k"
      >
        k
      </button>
      <button
        class="letterButton"
        id="l"
        value="l"
      >
        l
      </button>
      <button
        class="letterButton"
        id="m"
        value="m"
      >
        m
      </button>
      <button
        class="letterButton"
        id="n"
        value="n"
      >
        n
      </button>
      <button
        class="letterButton"
        id="o"
        value="o"
      >
        o
      </button>
      <button
        class="letterButton"
        id="p"
        value="p"
      >
        p
      </button>
      <button
        class="letterButton"
        id="q"
        value="q"
      >
        q
      </button>
      <button
        class="letterButton"
        id="r"
        value="r"
      >
        r
      </button>
      <button
        class="letterButton"
        id="s"
        value="s"
      >
        s
      </button>
      <button
        class="letterButton"
        id="t"
        value="t"
      >
        t
      </button>
      <button
        class="letterButton"
        id="u"
        value="u"
      >
        u
      </button>
      <button
        class="letterButton"
        id="v"
        value="v"
      >
        v
      </button>
      <button
        class="letterButton"
        id="w"
        value="w"
      >
        w
      </button>
      <button
        class="letterButton"
        id="x"
        value="x"
      >
        x
      </button>
      <button
        class="letterButton"
        id="y"
        value="y"
      >
        y
      </button>
      <button
        class="letterButton"
        id="z"
        value="z"
      >
        z
      </button>
    </p>
  </div>
</div>
`;
